pulsar/impl_partition_producer.go

```diff
@@ -293,14 +293,17 @@ func (p *partitionProducer) internalFlushCurrentBatch() {
 func (p *partitionProducer) internalFlush(fr *flushRequest) {
 	p.internalFlushCurrentBatch()
 
-	pi := p.pendingQueue.PeekLast().(*pendingItem)
```

Comment:
In point of my view, if type is specific, meaning the item from pendingQueue is *pendingItem, we can not type assert with ok.

Reply:
> In point of my view, if type is specific, meaning the item from pendingQueue is *pendingItem, we can not type assert with ok.

As [comment on PeekLast](https://github.com/apache/pulsar-client-go/blob/7a9aa2b36371437b1c409cb21cb2056ded7f198d/util/blocking_queue.go#L40)

> // PeekLast return last item in queue without dequeing, return nil if queue is empty

PeekLast() returns nil if queue is empty.

panic happens when ```PeekLast``` returns nil